mlos_bench/mlos_bench/optimizers/base_optimizer.py

```diff
@@ -41,8 +41,8 @@ def __init__(self, tunables: TunableGroups, service: Optional[Service], config:
         self._tunables = tunables
         self._service = service
         self._iter = 1
-        self._use_defaults: bool = bool(strtobool(str(self._config.pop('use_defaults', True))))
-        self._max_iter = int(self._config.pop('max_iterations', 25))
+        self._use_defaults: bool = bool(strtobool(str(self._config.pop('use_defaults', False))))
```

Comment:
I think a desirable thing is for us to check the "default" values in the ConfigSpace as the very first experiment Trial.
Leaving this as `True` helps accomplish that

Reply:
Also, this is maybe just a confusion for me from the PR title, but leaving the defaults for unspecified initialization is then not a "random initialization" - it's pretty explicit in the configs, no?

Reply:
I've fixed it in the subsequent commits - will mark this conversation resolved for now